FS19/ML/Zusammenfassung_ML.tex

```diff
@@ -2480,4 +2482,123 @@ \subsubsection{Building a Random Forest}
+\end{itemize}
+
+\subsubsection{Under- and Overfitting in Regression}
+
+\begin{figure}[tbh!]
+    \centering
+    \includegraphics[width=0.7\linewidth, keepaspectratio]{Pictures/underfitting_overfitting_regression}
+    \caption{Underfitting and Overfitting in Regression}
+    \label{fig:underfittingoverfittingregression}
+\end{figure}
+
+\vspace{1em}
+
+\noindent
+\textbf{Underfitting}
+\begin{itemize}
+    \item A straight line is just not complex enough to approximate the data
+    \item This model performs badly on both training and test data
+\end{itemize}
+
+\noindent
+\textbf{Overfitting}
+\begin{itemize}
+    \item This model optimizes to much with respect to the training data
+    \item It shows high performance on training but low performance on test data
+\end{itemize}
+
+\clearpage
+
+\subsubsection{Learning Curve for Underfitting}
+     
+\begin{figure}[h!]
+    \centering
+    \includegraphics[width=0.7\linewidth, keepaspectratio]{Pictures/underfitting_learning_curve}
+    \caption{Underfitting has bad performance on training and test data}
+    \label{fig:underfittinglearningcurve}
+\end{figure}
+
+\subsubsection{Counteracting Underfitting}
+Underfitting is often easier to repair, as it means that your model is misses relevant structures, so you can try to make it more complex through:
+\begin{itemize}
+    \item Try a larger set of features
+    \item Try a different set of features
+    \item Try more complex machine learning algorithms
+\end{itemize}
+
+\subsubsection{Learning Curve for Overfitting}
+
+\begin{figure}[h!]
+    \centering
+    \includegraphics[width=0.7\linewidth, keepaspectratio]{Pictures/overfitting_learning_curve}
+    \caption{Overfitting has good performance on training but bad performance on test data}
+    \label{fig:overfittinglearningcurve}
+\end{figure}
+
+\subsubsection{Counteracting Overfitting}
+Overfitting happens more frequently and is much harder to repair. It occurs cause your model is to sensitive to the training data, you could try to:
+\begin{itemize}
+    \item Try getting more training data
+    \item Try better data cleaning and eliminate outliers
+    \item Try a smaller set of features through feature selection
+    \item Try early stopping during iterative training
+    \item Try tuning the regularization parameter
+    \item Try ensemble methods
+\end{itemize}
+
+\subsubsection{Early Stopping}
+
+\begin{figure}[h!]
+    \centering
+    \includegraphics[width=0.6\linewidth, keepaspectratio]{Pictures/early_stopping}
+    \caption{Stopping the training early to avoid ovefitting}
+    \label{fig:earlystopping}
+\end{figure}
+
+\subsection{How to Approach a Machine Learning Project}
+
+\begin{enumerate}
+    \item Data quality assessment is always worth the time
+    \item Visualize your data
+    \item Specify and implement a water-proof quality assessment workflow
+    \item Implement a quick-and-dirty baseline classifier or regressor
```

Comment:
Add the example with the baseline classifier from the lecture

Reply:
I did not write that down